syntax = "proto3";

package inventory;
enum SelectOption {
    GET = 0;
    GET_ALL = 1;
    ADD = 2;
    REDUCE=4;
    CHECK = 5;
    UPDATE = 3;
    DELETE = 6;
    CREATE=7;
    PAYMENT_DONE=8;
    CHECK_BY_PRODUCT=9;

}

message Inventory {
    int32 id =1;
    string product_id=2;
    string inventory_id =3;
    int64 stock_level = 4;
    int64 add_stock_level = 5;
    int64 reduce_stock_level = 6;
    int64 reserved_stock = 7;
    string error_message = 12;
    SelectOption option =13;
    int64 http_status_code = 14;
    int64 sold_stock=15;
}

message Product {
    int32 id = 1;
    string product_id = 2;
    string name = 3;
    string description = 4;
    float price = 5;
    bool is_available = 6;
    string message = 7;
    string error_message = 8;
    SelectOption option = 9;
    int64 http_status_code = 10;
}

message Order {
    int32 id =1;
    string order_id=2;
    string product_id =3;
    string user_id = 4;
    string username =5;
    string email=6;
    int64 quantity= 7;
    string shipping_address = 8;
    string customer_notes = 9;
    int64 stock_level = 10;
    bool is_product_available = 11;
    bool is_stock_available  = 12;
    string message = 13;
    string error_message = 14;
    OrderStatus order_status = 15;
    PaymentStatus payment_status = 16;
    SelectOption option =17;
    int64 http_status_code = 18;
}
enum OrderStatus {
  IN_PROGRESS = 0;
  COMPLETED = 1;
}
enum PaymentStatus {
    PENDING = 0;
    PAID = 1;
    FAILED = 2;
}

message InventoryItem {
    repeated Inventory inventories = 1;
}
